### PR TITLE
feat: inform that a new version is available

### PR DESCRIPTION
**Pull request by Adaendra**
### Related PR
- https://github.com/RoadTripMoustache/kana_to_kanji/pull/259

## 📖 Description
Implementation of upgrader to inform the user that a new version of the application is out and help them to get it easily.

### ⁉️ Related Issues
closes: #29 

### 🖼️ Screenshots:

<details>
    <summary>Use case</summary> 

[issue_29.webm](https://github.com/RoadTripMoustache/kana_to_kanji/assets/36586573/30a2f1a0-6956-4eff-bfbf-2623387e6f18)

</details>

### 🧪 How to test the change?
- Add the following code somewhere in the app
```dart
class MyUpgrader extends Upgrader {
  MyUpgrader({super.debugLogging, super.storeController});

}
```
- Add the following code in the `build` method of `prepare_quiz-view.dart`
```dart
final upgrader = MyUpgrader(
      debugLogging: true,
      storeController: UpgraderStoreController(
        onAndroid: () => UpgraderAppcastStore(
            appcastURL:
                'https://raw.githubusercontent.com/larryaasen/upgrader/master/test/testappcast.xml'),
        oniOS: () => UpgraderAppcastStore(
            appcastURL:
                'https://raw.githubusercontent.com/larryaasen/upgrader/master/test/testappcast.xml'),
      ),
    );
```
- Add `upgrader = upgrader` in the `AppSnackBarUpgrader` declaration  in the `build` method of `prepare_quiz-view.dart`
- Open the application

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] I have added/updated tests.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`.

### CODE REVIEW

File: localization/en.arb [hunk 1]
```diff
@@ -307,5 +307,16 @@
   },
   "glossary_sort_by_title": "Filter by",
   "glossary_sort_by_alphabetical": "Alphabetical (A-Z)",
-  "glossary_sort_by_japanese": "Japanese syllable (あ-お)"
+  "glossary_sort_by_japanese": "Japanese syllable (あ-お)",
+  "snackbar_update_version_action": "Update",
+  "snackbar_update_version_message": "Version {version} is now available !",
```
Comment (apomalyn) [inline localization/en.arb:312]: Pour facilité la compréhension du message mieux vaut: 
```suggestion
  "snackbar_update_version_message": "A new version is available !",
```

File: localization/fr.arb [hunk 2]
```diff
@@ -302,5 +302,16 @@
   },
   "glossary_sort_by_title": "Filtrer par",
   "glossary_sort_by_alphabetical": "Alphabétique (A-Z)",
-  "glossary_sort_by_japanese": "Syllabe japonaise (あ-お)"
+  "glossary_sort_by_japanese": "Syllabe japonaise (あ-お)",
+  "snackbar_update_version_action": "Mettre à jour",
+  "snackbar_update_version_message": "La version {version} est disponible !",
```
Comment (apomalyn) [inline localization/fr.arb:307]:
```suggestion
  "snackbar_update_version_message": "Une nouvelle version est disponible !",
```

File: lib/src/quiz/prepare/prepare_quiz_view.dart [hunk 3]
```diff
@@ -17,7 +18,8 @@ class PrepareQuizView extends StatelessWidget {
   Widget build(BuildContext context) {
     final l10n = AppLocalizations.of(context);
 
-    return ViewModelBuilder<PrepareQuizViewModel>.reactive(
+    return AppSnackBarUpgrader(
```
Comment (apomalyn) [inline lib/src/quiz/prepare/prepare_quiz_view.dart:21]: Je le mettrais autour de `KanaToKanjiApp` pour permettre de l'avoir quelque soit le point d'entrée dans l'app. Voir la [doc](https://pub.dev/packages/upgrader#go-router) pour GoRouter

**Review by apomalyn — CHANGES_REQUESTED** [review]

File: lib/src/quiz/prepare/prepare_quiz_view.dart (hunk 3 above)
Comment (Adaendra) [reply lib/src/quiz/prepare/prepare_quiz_view.dart:21]: Changement fait

File: localization/en.arb [hunk 4]
```diff
@@ -307,5 +307,7 @@
   },
   "glossary_sort_by_title": "Filter by",
   "glossary_sort_by_alphabetical": "Alphabetical (A-Z)",
-  "glossary_sort_by_japanese": "Japanese syllable (あ-お)"
+  "glossary_sort_by_japanese": "Japanese syllable (あ-お)",
+  "snackbar_update_version_action": "Update",
+  "snackbar_update_version_message": "A new version is available !"
```
Comment (apomalyn) [inline localization/en.arb:312]: En anglais, il n'y a pas d'espace avec un point d'exclamation ou d'interrogation 
```suggestion
  "snackbar_update_version_message": "A new version is available!"
```

**Review by apomalyn — DISMISSED** [review]
Une dernière petite modification mais tu as mon approbation pour la merge quand tu l'as fait!

**Review by apomalyn — APPROVED** [review]

**Pull request merged**